### ISSUE TITLE
[HOLD #373588][$500] IOU - Unpaid report title changed to "paid" after workspace has been deleted

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#38339]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.52-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4425785
**Issue reported by:** Applause - Internal Team

## Action Performed:
Pre-requisite: the user must be logged in and have created a workspace.
1. Go to workspace chat.
2. Create a money request.
3. Open the IOU report.
4. Verify the IOU title in the header is "[Workspace] owes [amount]".
5. Delete the workspace.
6. Go to workspace archived chat.
7. Open the IOU report.
8. Verify the IOU title is now changed to "[Workspace] paid [amount]".

## Expected Result:
The unpaid IOU title should remain the same after workspace deletion.

## Actual Result:
The unpaid IOU title is changed to "[Workspace] paid [amount]" after workspace deletion.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/a6091033-f361-4284-8504-9a2ed804219c

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a4106c0d7032d428</li>
        <li>Upwork Job ID: 1769817849588871168</li>
        <li>Last Price Increase: 2024-04-01</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
@slafortune I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by slafortune** [comment]
For clear accounting, I think it's important to not change the status to Paid when in fact an IOU is unpaid.

**Comment by kmbcook** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
IOU - Unpaid report title changed to "paid" after workspace has been deleted

### What is the root cause of that problem?
When deleting a workspace, the backend overwrites needed information previously stored in report.stateNum and report.statusNum.

### What changes do you think we should make in order to solve the problem?
Fix the backend.

**Comment by mountiny** [comment]
@srikarparsi will be making holistic changes to archiving reports. Would you want to put this issue on hold for your project?

**Comment by srikarparsi** [comment]
Hi @shubham1206agra, I talked to @mountiny offline and my holistic changes will not cover this. We should fix this separately so no need to hold.

**Comment by Tony-MK** [comment]
Just curious,

From this https://github.com/Expensify/App/issues/36371#issuecomment-1997959627, I just wanted to confirm if we should archive IOUs offline or not.

Thanks.

**Comment by srikarparsi** [comment]
We actually changed the behavior in the project based on this [conversation](https://expensify.slack.com/archives/C036QM0SLJK/p1712049362474879) so now the wholistic changes will cover this bug. Putting this issue on hold for this tracking [issue](https://github.com/Expensify/Expensify/issues/373588)

**Comment by slafortune** [comment]
Still on hold and may be fixed as addressed [here](https://github.com/Expensify/App/issues/38339#issuecomment-2033515617) moving to weekly.

**Comment by slafortune** [comment]
On hold - not complete yet

**Comment by slafortune** [comment]
Still on hold

**Comment by slafortune** [comment]
Looks like they plan for this to be complete soon but still on hold for tracking https://github.com/Expensify/Expensify/issues/373588

@srikarparsi I will be out on vacation until June 4th and am not assigning another to this GH, it doesn't appear that any action will need to be taken.  Can you please provide an update as well?  TIA

**Comment by slafortune** [comment]
[Update](https://github.com/Expensify/Expensify/issues/373588#issuecomment-2144457518)
> Working on implementation, was OOO last week. Updated ETA for completing code is June 7th and am aiming to run the migration by June 12th.

**Comment by slafortune** [comment]
Still on hold

**Comment by slafortune** [comment]
Still on hold - no update

**Comment by slafortune** [comment]
still on hold

**Comment by slafortune** [comment]
still on hold

**Comment by slafortune** [comment]
@shubham1206agra  - @srikarparsi will have an update by the end of the week - 
Adding another BZ in the meantime - I'll be out until 8/21 and will check in on this then.

**Comment by lschurr** [comment]
This seems to be on hold - Is that correct @srikarparsi?

**Comment by lschurr** [comment]
Assigning you here since you were mentioned earlier in the issue - Do you know what update we're waiting on for this one @srikarparsi?

**Comment by srikarparsi** [comment]
Ah sorry for missing your last bump. I believe this should've been fixed. I'll test it out right now to check.

**Comment by srikarparsi** [comment]
This has been fixed, we now title the expense report with the reportID so it doesn't show the paid/owe text. We also no longer change the state when this happens so we shouldn't see this bug in the future.
